git-fixup

```diff
@@ -67,6 +69,19 @@ function call_commit() {
     git commit ${git_commit_args[@]} --$op=$target
 }
 
+# Call git rebase
+function call_rebase() {
+    local target=$1
+
+    has_parent=`git rev-parse --quiet --verify $target~1^{commit}`
+    ret_has_parent=$?
+    if [ $ret_has_parent -eq 0 ]; then
```

Comment:
> to check the exit code you use the program directly in the if

Thanks!

Reply:
> should the refspec below also have ^{commit} what does that change?

I couldn't find a straightforward way to see if a sha is a valid commit. I found [this question](https://stackoverflow.com/questions/18515488/how-to-check-if-the-commit-exists-in-a-git-repository-by-its-sha-1) where solutions alternate between using `git cat-file` and `git rev-parse`. 

I eventually decided for `git rev-parse` which checks if the sha supplied is a _valid sha_ instead of being a valid commit inside the repo. By adding the `^{commit}` I ensure that the commit lives inside the repo:

```
$ git rev-parse --verify 1234567890123456789012345678901234567890
1234567890123456789012345678901234567890
$ git rev-parse --verify 1234567890123456789012345678901234567890^{commit}
fatal: Needed a single revision
```